tests/test_packets.py

```diff
@@ -40,16 +49,25 @@ def test_ensure_all_packets(protocol, config):
             for i in range(0, c[1]):  # iterate through packet ids which should be there
                 if i not in packet_map.packets[state].server_bound:
                     missing_serverbound.append(f"0x{i:02X}")
-        
+
         if (missing_serverbound or missing_clientbound) and initial_check:
             initial_check = False
             print()
 
         if missing_clientbound:
-            print(f"{colorama.Style.BRIGHT}{colorama.Fore.RED}MISSING (state={state.name}, CLIENT-BOUND):{colorama.Style.RESET_ALL}", ", ".join(missing_clientbound))
+            print(
+                f"{colorama.Style.BRIGHT}{colorama.Fore.RED}MISSING (state={state.name}, CLIENT-BOUND):{colorama.Style.RESET_ALL}",
+                ", ".join(missing_clientbound),
+            )
 
         if missing_serverbound:
-            print(f"{colorama.Style.BRIGHT}{colorama.Fore.RED}MISSING (state={state.name}, SERVER-BOUND):{colorama.Style.RESET_ALL}", ", ".join(missing_serverbound))
+            print(
+                f"{colorama.Style.BRIGHT}{colorama.Fore.RED}MISSING (state={state.name}, SERVER-BOUND):{colorama.Style.RESET_ALL}",
+                ", ".join(missing_serverbound),
+            )
 
     if initial_check:
         print(f"{colorama.Style.BRIGHT}{colorama.Fore.GREEN}All packets present!")
+    else:
+        reason = "Missing packets\n" + capsys.readouterr().out
```

Comment:
What does this do? Missing packets shouldn't make the tests fail.

Reply:
it doesnt. thats why its an xfail, which still passes

====================================================================== test session starts =======================================================================
platform linux -- Python 3.9.6, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/the456gamer/projects/pymine/PyMine-Net, configfile: pyproject.toml, testpaths: tests
collected 20 items                                                                                                                                               

tests/test_buffer.py .................                                                                                                                     [ 85%]
tests/test_nbt.py ..                                                                                                                                       [ 95%]
tests/test_packets.py x                                                                                                                                    [100%]

============================================================================= PASSES =============================================================================
==================================================================== short test summary info =====================================================================
XFAIL tests/test_packets.py::test_ensure_all_packets[757-config0]
  reason: Missing packets
PACKET CHECK (protocol=757): 
MISSING (state=LOGIN, CLIENT-BOUND): 0x00, 0x01, 0x02, 0x03
MISSING (state=LOGIN, SERVER-BOUND): 0x00, 0x01
MISSING (state=PLAY, CLIENT-BOUND): 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E, 0x2F, 0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x3F, 0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48, 0x49, 0x4A, 0x4B, 0x4C, 0x4D, 0x4E, 0x4F, 0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57, 0x58, 0x59, 0x5A, 0x5B, 0x5C, 0x5D, 0x5E, 0x5F, 0x60, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66
MISSING (state=PLAY, SERVER-BOUND): 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17, 0x18, 0x19, 0x1A, 0x1B, 0x1C, 0x1D, 0x1E, 0x1F, 0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2A, 0x2B, 0x2C, 0x2D, 0x2E

================================================================= 19 passed, 1 xfailed in 0.10s ==================================================================
all other tests still run

Reply:
But what is the point?